src/app/modules/item/components/item-children-edit-form/item-children-edit-form.component.ts

```diff
@@ -39,7 +39,9 @@ export class ItemChildrenEditFormComponent implements PendingChangesComponent, O
     private itemDataSource: ItemDataSource,
     private actionFeedbackService: ActionFeedbackService,
     private pendingChangesService: PendingChangesService,
-  ) {
+  ) {}
+
+  ngOnInit(): void {
     this.pendingChangesService.set(this);
   }
```

Comment:
Is this a bugfix, was there something wrong?
I'm asking because I have to investigate on something triggering [this error](https://github.com/France-ioi/AlgoreaFrontend/blob/master/src/app/shared/guards/pending-changes-guard.ts#L34), does this piece of code fixes that?

Reply:
Yes, the bug was next: 

The constructor() runs before lifecycle hooks, so when you go from ComponentA with pending changes to ComponentB with pending changes, the constructor in ComponentB runs pendingChangesService.set before the ComponentA runs ngOnDestroy which is does clear pendingChangesService. So I guess this is exactly your case you want investigate.